File: selenium3-java/src/test/java/starhotel/pages/TopPage.java
package starhotel.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class TopPage {

  private WebDriver driver;

  public TopPage(WebDriver driver) {
    this.driver = driver;
    if (!this.driver.getTitle().equals("STAR HOTEL - テスト自動化デモサイト")) {
      throw new IllegalStateException("現在のページが間違っています: " + this.driver.getTitle());
    }
  }

  public LoginPage goToLoginPage() {
    var loginLink = driver.findElement(By.linkText("ログイン"));
    loginLink.click();
    return new LoginPage(driver);
  }

  public SignupPage goToSignupPage() {
    var signupLink = driver.findElement(By.linkText("会員登録"));
    signupLink.click();
    return new SignupPage(driver);
  }

  public PlansPage goToPlansPage() {
    var planLink = driver.findElement(By.linkText("宿泊予約"));
    planLink.click();
    return new PlansPage(driver);
  }
}

File: webdriverio/test/pageobjects/top.page.js
const Page = require('./page');

class TopPage extends Page {
  get loginLink() { return $('=ログイン'); }
  get signupLink() { return $('=会員登録'); }
  get planLink() { return $('=宿泊予約'); }

  open() {
    super.open('');
  }

  goToLoginPage() {
    this.loginLink.click();
  }

  goToSignupPage() {
    this.signupLink.click();
  }

  goToPlansPage() {
    this.planLink.click();
  }
}

module.exports = new TopPage();
